### ISSUE TITLE
dinit support?

### ISSUE THREAD

**Issue by ManjTi** [fairyglade/ly#528]
OpenRC & runit is working, but can't make it work with dinit.

**Comment by jackffmm** [comment]
I'd love that!

**Comment by procobain** [comment]
+1

**Comment by Daxik2x** [comment]
yes pls

**Comment by spflaumer** [comment]
Hello guys,
I have been playing around with dinit for a bit now and finally managed to get ly "working" under it.

It seems that simply stopping the `tty1` service and removing it from `/etc/dinit.d/` directory doesn't actually free `/dev/tty1`.
Instead, dinit marks the consoles as "active" within `/etc/dinit.d/config/console.conf`.
What that option means exactly I'm not sure.

By changing the `ACTIVE_CONSOLES` option from "/dev/tty[1-6]" to "/dev/tty[2-6]" so that tty1 becomes free, `ly` starts successfully.

Note: for some reason, the `tty` option within `/etc/ly/config.ini` is ignored and only tty1 is ever used.

TL;DR:
Follow the instructions for dinit on my fork [ly-dinit](https://github.com/spflaumer/ly-dinit)

OR

1. remove `/etc/dinit.d/tty1`
2. open `/etc/dinit.d/config/console.conf` and change `ACTIVE_CONSOLES="/dev/tty[1-6]"` to `ACTIVE_CONSOLES="/dev/tty[2-6]"`
3. create `/etc/dinit.d/ly` with the following content:
```
type = process
restart = true
smooth-recovery = true
# note: /usr/bin/ly-dm when installing from pacman, /usr/bin/ly when building from source
command = /usr/bin/ly-dm
depends-on = loginready
termsignal = HUP
# since loginready already occupies the console, this option allows to still access it
options = shares-console
```
4. enable the service: `sudo dinitctl enable ly`
5. after rebooting ly should appear on tty1

**Comment by javalsai** [comment]
> Hello guys,
> I have been playing around with dinit for a bit now and finally managed to get ly "working" under it.
> 
> It seems that simply stopping the `tty1` service and removing it from `/etc/dinit.d/` directory doesn't actually free `/dev/tty1`.
> Instead, dinit marks the consoles as "active" within `/etc/dinit.d/config/console.conf`.
> What that option means exactly I'm not sure.
> 
> By changing the `ACTIVE_CONSOLES` option from "/dev/tty[1-6]" to "/dev/tty[2-6]" so that tty1 becomes free, `ly` starts successfully.
> 
> Note: for some reason, the `tty` option within `/etc/ly/config.ini` is ignored and only tty1 is ever used.
> 
> TL;DR:
> Follow the instructions for dinit on my fork [ly-dinit](https://github.com/spflaumer/ly-dinit)
> 
> OR
> 
> 1. remove `/etc/dinit.d/tty1`
> 2. open `/etc/dinit.d/config/console.conf` and change `ACTIVE_CONSOLES="/dev/tty[1-6]"` to `ACTIVE_CONSOLES="/dev/tty[2-6]"`
> 3. create `/etc/dinit.d/ly` with the following content:
> ```
> type = process
> restart = true
> smooth-recovery = true
> # note: /usr/bin/ly-dm when installing from pacman, /usr/bin/ly when building from source
> command = /usr/bin/ly-dm
> depends-on = loginready
> termsignal = HUP
> # since loginready already occupies the console, this option allows to still access it
> options = shares-console
> ```
> 4. enable the service: `sudo dinitctl enable ly`
> 5. after rebooting ly should appear on tty1
> 

Amazing!! I was messing with the dinit service too a while ago but missed thr active consoles thing. Now it works! But right after boot, it automatically switches to tty2 and I have to manually go back to tty1, do you know about a way to be in tty1 automatically? Maybe a console conf too?

**Comment by davmac314** [comment]
Hi, Dinit maintainer here, just wanted to note some things:

> Instead, dinit marks the consoles as "active" within /etc/dinit.d/config/console.conf.

Just to be clear, that's not actually part of dinit, it's something that the distribution is doing. (Looking art Artix scripts, I would've thought removing `/etc/dinit.d/tty1` would be enough regardless, but maybe the tty services are being dynamically recreated).

>     # since loginready already occupies the console, this option allows to still access it
>     options = shares-console

It might not matter too much, but my guess would be that ly should run on a specific tty, not on the console. So it should be run via `getty` (or `agetty`). Something like:

    command = /sbin/agetty -n -l /usr/bin/ly-dm tty1

Then, you should not need the `shares-console` option. That might help with the automatic switching to tty2 mentioned above by @javalsai .

**Comment by javalsai** [comment]
> Hi, Dinit maintainer here, just wanted to note some things:
> 
> > Instead, dinit marks the consoles as "active" within /etc/dinit.d/config/console.conf.
> 
> Just to be clear, that's not actually part of dinit, it's something that the distribution is doing. (Looking art Artix scripts, I would've thought removing `/etc/dinit.d/tty1` would be enough regardless, but maybe the tty services are being dynamically recreated).
> 
> > ```
> > # since loginready already occupies the console, this option allows to still access it
> > options = shares-console
> > ```
> 
> It might not matter too much, but my guess would be that ly should run on a specific tty, not on the console. So it should be run via `getty` (or `agetty`). Something like:
> 
> ```
> command = /sbin/agetty -n -l /usr/bin/ly-dm tty1
> ```
> 
> Then, you should not need the `shares-console` option. That might help with the automatic switching to tty2 mentioned above by @javalsai .

Sorry, wanted to test this and answer a while back, but I've been busy, that command did in fact make the console parameters unnecessary, but it still switches to tty2. But I don't have much time today to keep tinkering around, and I feel like I won't feel like trying it tomorrow unless I promise to.

What I just did so far is mimic the config of tty2, that spawns the command `/usr/lib/dinit/agetty-default tty2`, so I just copied the script and replaced exec with echo, the rest is just config and argument parsing. That just got me to the command `/usr/bin/agetty tty2 38400 linux`, so maybe I just have to pass such arguments to agetty to prevent the switch?

**Comment by AnErrupTion** [comment]
Implemented in master. Also Ly doesn't ignore the TTY number anymore.
